bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/filter/BigtableFilter.java

```diff
@@ -69,17 +67,17 @@ public boolean equals(Object obj) {
       return false;
     }
     BigtableFilter other = (BigtableFilter) obj;
-    return rowFilter.equals(other.rowFilter);
+    return filter.toProto().equals(other.filter.toProto());
   }
 
   @Override
   public byte[] toByteArray() throws IOException {
-    return rowFilter.toByteArray();
+    return filter.toProto().toByteArray();
   }
 
   public static BigtableFilter parseFrom(final byte[] bytes) throws DeserializationException {
     try {
-      return new BigtableFilter(RowFilter.parseFrom(bytes));
+      return new BigtableFilter(FILTERS.fromProto(RowFilter.parseFrom(bytes)));
```

Comment:
Updated `BigtableFilter` with `models.Filters.Filter`, 
 
Here I am relaying on `FILTERS#fromProto` for creating a `models.Filters.Filter` using deserialization. **Is this approach correct?** or Are there any other ways to create a `models.Filters.Filter` using deserialization? I looked but didn't found one.

@sduskis Could you please have a look and let me know your thought on this.

Reply:
This sounds like the correct approach for now.

@igorbernstein2, we should consider making all `Filters.Filter` serializable in the future.